rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -1560,6 +1674,28 @@ Player::Player(
     storage_options, play_options, node_name, node_options)
 {}
 
+Player::Player(
+  const std::vector<rosbag2_storage::StorageOptions> & storage_options,
+  const rosbag2_transport::PlayOptions & play_options,
+  const std::string & node_name,
+  const rclcpp::NodeOptions & node_options)
+: rclcpp::Node(
+    node_name,
+    rclcpp::NodeOptions(node_options).arguments(play_options.topic_remapping_options))
+{
+  std::shared_ptr<KeyboardHandler> keyboard_handler;
+  if (!play_options.disable_keyboard_controls) {
+    keyboard_handler = std::make_shared<KeyboardHandler>();
+  }
+
+  std::vector<reader_storage_options_pair_t> input_bags{};
+  for (const auto & options : storage_options) {
+    input_bags.emplace_back(std::make_unique<rosbag2_cpp::Reader>(), options);
+  }
+  pimpl_ = std::make_unique<PlayerImpl>(
```

Comment:
```suggestion
  std::vector<reader_storage_options_pair_t> input_bags{};
  for (const auto & options : storage_options) {
    input_bags.emplace_back(ReaderWriterFactory::make_reader(options), options);
  }
  pimpl_ = std::make_unique<PlayerImpl>(
```
Also, please rename `input_bags` to the `readers_with_options` for clarity

Reply:
changed in 82814f011d9f938889bc50d4d9060b95b1de0689